### ISSUE TITLE
PrivateKey.public cannot be used in the promise interface

### ISSUE THREAD

**Issue by sloonz** [sjudson/paseto.js#28]
I use this code to generate a keypair : 

```javascript
const Paseto = require('paseto.js');    
    
async function main() {    
  const priv = new Paseto.PrivateKey(new Paseto.V2());    
  await priv.generate();    
  const pub = await priv.public();                                                      
  console.log(`Private key: ${(await priv.raw()).toString('base64')}`);    
  console.log(`Public key: ${(await pub.raw()).toString('base64')}`);      
}                                                                        

main().catch(err => {    
  console.log(err);      
});                    
```

Unfortunately, this fails because `priv.public()` returns undefined instead of a promise. The callback-style API works fine.

**Comment by sjudson** [comment]
Thanks for the report, will look at and hopefully get a fix up over the weekend.

**Comment by sjudson** [comment]
Fixed in `0.1.6`.

**Comment by sloonz** [comment]
Thank you !
